차한솔/week2/베스트앨범.js

```diff
@@ -0,0 +1,33 @@
+function solution(genres, plays) {
+	const allGenresWithPlays = new Map();
+	
+	for (let i = 0; i < genres.length; i++) {
+		const genre = genres[i];
+
+		if (!allGenresWithPlays.has(genre)) {
+			allGenresWithPlays.set(genre, {
+				totalPlays: plays[i],
+				songs: [{index: i, plays: plays[i]}],
+			})
+
+			continue;
+		}
+
+		const genreWithPlay = allGenresWithPlays.get(genre)
+
+		allGenresWithPlays.set(genre, {
+			totalPlays: genreWithPlay.totalPlays + plays[i],
+			songs: [...genreWithPlay.songs, {index: i, plays: plays[i]}]
+		})
```

Comment:
현재 key가 존재 하냐 하지않냐를 기준으로 두 가지 경우에 분기되어 로직이 작성되는데 early 리턴이 유용한 케이스가 아닌 거 같습니다.
```suggestion
		if (!allGenresWithPlays.has(genre)) {
			allGenresWithPlays.set(genre, {
				totalPlays: plays[i],
				songs: [{index: i, plays: plays[i]}],
			})

			continue;
		} else {
			const genreWithPlay = allGenresWithPlays.get(genre)

		allGenresWithPlays.set(genre, {
			totalPlays: genreWithPlay.totalPlays + plays[i],
			songs: [...genreWithPlay.songs, {index: i, plays: plays[i]}]
		})
		}
```
위와 같이 if...else 문으로 작성해 보는 것은 어떨까요?